x-pack/plugins/ml/server/models/job_audit_messages/job_audit_messages.js

```diff
@@ -126,18 +131,25 @@ export function jobAuditMessagesProvider({ asInternalUser }, mlClient) {
     });
 
     let messages = [];
+    const notificationIndices = [];
+
     if (body.hits.total.value > 0) {
-      messages = body.hits.hits.map((hit) => ({
-        clearable: hit._index === ML_NOTIFICATION_INDEX_02,
-        ...hit._source,
-      }));
+      let notificationIndex;
+      messages = body.hits.hits.map((hit) => {
+        if (notificationIndex !== hit._index && isClearable(hit._index)) {
+          notificationIndices.push(hit._index);
+          notificationIndex = hit._index;
+        }
+
+        return hit._source;
+      });
```

Comment:
`map` is intended to use to create a new collection, but there you have a side effect of updating the `notificationIndices ` array. I think for readability it'd be better just to have a `forEach` that populates messages and notification indicies.

Reply:
Agreed - updated for better readability in dfe7484fb36ec61d27085d18e60a46a2807d79df